### ISSUE TITLE
Postgres support for using Array for select prop in (?) queries

### ISSUE THREAD

**Issue by rbygrave** [ebean-orm/ebean#1186]
This can improve query plan caching as we have variations on the SQL relative to many ? parameters.

**Comment by rbygrave** [comment]
I am putting this on hold as the query execution plans in Postgres indicate that this isn't as good as the simple in (?,?,? ...) option.

We need to put more effort into reviewing the resulting query plans for this before this can be merged etc.

**Comment by rbygrave** [comment]
@rPraml I have noted that the query execution plans with this don't look as good in Postgres.  I'll put this on hold and not merge in until I can review it in more detail.

Also note that this only has the H2 and Postgres support added (and not SqlServer and Oracle).

Cheers, Rob.

**Comment by rbygrave** [comment]
With Postgres ... `= ANY` is giving better query plans so that is the path to follow for this.  We need to be careful with `not in` as `!= ALL`.

**Comment by rbygrave** [comment]
Leaving this as Postgres only using PG `= ANY` and `!= ALL`

**Comment by rbygrave** [comment]
Yes, all good. I saw your branch changes and this is all based off that.  I saw the null detection/handling for in() ... so I'm pondering that (I don't really expect null handling in an IN clause as I see NULL as pretty special so I'm pondering that part and excluded it from this change for now).

This is only restricted to Postgres use at this stage.

To get the ANY we need to also pass the 'not' as ANY becomes != ALL in the 'not in' case (so that is what I needed to change to get use of Postgres ANY ... but yes, I am not convinced by the query plans I have seen thus far with select unnest so I'm forging ahead with `= ANY(?)` and `!= ALL(?)`.  In fact I think I pushed that into master.

But yes, I understand you are busy so no worries if you don't read/consume this until later etc.
